src/app/login/login.component.ts

```diff
@@ -43,12 +43,16 @@ export class LoginComponent {
 
     createMode:boolean = this.router.url.split('?')[0] === '/login/newuser';
     returnUrl = this.route.snapshot.queryParams['returnUrl'] || '/';
-    model = { name:'', password:'', repeatPassword:'' }
+    model = { name:'', password:'', repeatPassword:'' };
     message = '';
 
     onSubmit() {
         if(this.createMode) {
-            this.createUser(this.model);
+            if (this.checkAdminExistence()) {
+                this.createAdmin(this.model);
+            } else {
+                this.createUser(this.model);

```

Comment:
As written, the checkAdminExistence method actually doesn't return anything, so this will always be false.  The functions within a `.then()` chain don't return their value to a parent function, they return it to the next `.then()`.

As a note when you send off an async request with a promise like the `couchService.get()` that anything that needs to happen after needs to be within one of the `.then()` functions attached to the promise.